pkg/export/disk/disk.go

```diff
@@ -0,0 +1,314 @@
+package disk
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/export/util"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/logging"
+	"k8s.io/client-go/util/retry"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type Connection struct {
+	// path to store audit logs
+	Path string `json:"path,omitempty"`
+	// max number of audit results to store
+	MaxAuditResults int `json:"maxAuditResults,omitempty"`
+	// File to write audit logs
+	File *os.File
+
+	// current audit run file name
+	currentAuditRun string
+}
+
+type Writer struct {
+	openConnections map[string]Connection
+}
+
+const (
+	Name                = "disk"
+	maxAllowedAuditRuns = 5
+	maxAuditResults     = "maxAuditResults"
+	violationPath       = "path"
+)
+
+var Connections = &Writer{
+	openConnections: make(map[string]Connection),
+}
+
+var log = logf.Log.WithName("disk-driver").WithValues(logging.Process, "export")
+
+func (r *Writer) CreateConnection(_ context.Context, connectionName string, config interface{}) error {
+	cfg, ok := config.(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("invalid config format")
+	}
+
+	path, pathOk := cfg[violationPath].(string)
+	if !pathOk {
+		return fmt.Errorf("missing or invalid 'path' for connection %s", connectionName)
+	}
+	if err := validatePath(path); err != nil {
+		return fmt.Errorf("invalid path: %w", err)
+	}
+	maxResults, maxResultsOk := cfg[maxAuditResults].(float64)
+	if !maxResultsOk {
+		return fmt.Errorf("missing or invalid 'maxAuditResults' for connection %s", connectionName)
+	}
+	if maxResults > maxAllowedAuditRuns {
+		return fmt.Errorf("maxAuditResults cannot be greater than %d", maxAllowedAuditRuns)
+	}
+
+	r.openConnections[connectionName] = Connection{
+		Path:            path,
+		MaxAuditResults: int(maxResults),
+	}
+	return nil
+}
+
+func (r *Writer) UpdateConnection(_ context.Context, connectionName string, config interface{}) error {
```

Comment:
lots of common code can be shared between create and update connection.

Reply:
Cleaned up disk driver code, should be good now.